### DISCUSSION TITLE
Is this possible to run Trame Application as Standalone Backend Server

### DISCUSSION THREAD

**Discussion by phtran-dev** [Kitware/trame#196] [Q&A]
Hi there,
We are developing a 3D feature which can be integrated within OHIF. The problem is that OHIF is using React whilst Trame is using Vue. So we need to separate  backend server and client side. Is this possible to use Trame as Backend Server and develop a client side (using vtk.js) connecting to Trame ? Like in this remote rendering example : https://kitware.github.io/vtk-js/examples/RemoteView.html

**Comment by jourdain** [comment]
Indeed, trame can be used as just a server side implementation with any client side framework. The only thing (as you can imagine) is that you won't have the opportunity to define your Web client from Python. 

Technically trame is a VTK/ParaView Web application where the client is built-in. So for your use case, you can easily use vtk.js+wslink to drive your trame server like in the example you linked.

If you need help or guidance in that integration process, feel free to reach out to [Kitware for support](https://www.kitware.com/contact/).

**Comment by aylward** [comment]
Hi @phtran-dev 

We actually have a nearly complete solution for that already.   Take a look at the rendering and interface of Volview:
https://volview.netlify.app/
- note that this is pure javascript - no server (dicom or otherwise) is required.   It can be distributed as an html file :)   Just drag-and-drop your dicom files...and it supports many other file formats as well.

1) Next week we are releasing a version that integrates DICOMWeb capabilities as well as the ability to load data from a URL (e.g., hosted in the cloud).  If you are eager to test it now, I can provide you with a link to the pre-release version.  Email is best for this request: stephen.aylward@kitware.com

2) Within one month (worst-case two months) we will release yet another version that includes a python server.   As described by Seb, wslink is used for the communication between python and vtk.js in the app.   The server runs on the local machine by default, but you can also point VolView to other compliant python servers (e.g., running on a compute server on your network or running in the cloud).   In the demo server provided, it will run a MONAI-based segmentation system for CT data.  See https://monai.io and its model zoo for examples.

Here is a recent volview press release:
https://www.kitware.com/kitware-brings-3d-cinematic-rendering-to-your-web-browser-with-volview-4-0/

Looking forward to your feedback and questions.

Stephen
